docs/docs/data-sources/restapi.md

```diff
@@ -3,44 +3,150 @@ id: restapi
+- __Authentication Type__: The method of authentication to use with REST API requests. Supported Types: None, Basic, Bearer, and OAuth 2.0
+  - **Basic**: Requires Username and Password
+  - **Bearer**: Requires a token, typically a JSON Web Token (JWT), to grant access
+  - **OAuth 2.0**: The OAuth 2.0 protocol mandates the provision of the following parameters: access token URL, access token URL custom headers, client ID, client secret, scopes, custom query parameters, authorization URL, custom authentication parameters, and client authentication.
 
-- URL of the REST endpoint
+<div style={{textAlign: 'center'}}>
 
-The following optional parameters are also supported:
+<img className="screenshot-full" src="/img/datasource-reference/rest-api/oauth.png" alt="ToolJet - Data source - REST API" />
 
-   | Type         | Description |
-   | -----------  | ----------- |
-   | URL params   | Additional query string parameters|
-   | headers      | Any headers the REST API source requires|
-   | body         | Any values or fields the REST API source requires|
+</div>
 
 :::info
-REST HTTP methods that are supported are **GET, POST, PUT, PATCH &amp; DELETE**.
+REST HTTP methods that are supported are **GET, POST, PUT, PATCH & DELETE**.
 :::
 
-<img class="screenshot-full" src="/img/datasource-reference/rest-api/rest-api-values.gif" alt="ToolJet - Data source - REST API" height="420"/>
+<div style={{textAlign: 'center'}}>
+
+<img className="screenshot-full" src="/img/datasource-reference/rest-api/methods.png" alt="ToolJet - Data source - REST API" />
+
+</div>
 
 ## Querying REST API
-Click on `+` button of the query manager at the bottom panel of the editor and select the REST API endpoint added in the previous step as the data source.
 
-Click on the 'run' button to run the query. 
+Once you have connected to the REST API datasource, follow these steps to write queries and interact with a REST API in the ToolJet application:
 
-:::note
-NOTE: Query should be saved before running.
-:::
+
+1. Open the ToolJet application and navigate to the query panel at the bottom of the app builder.
+2. Click the `+Add` button to open the list of available `local` and `global datasources`.
+3. Select **REST API** from the global datasource section.
+4. Enter the required query parameters.
+5. Save the query.
+6. Click `Preview` to view the data returned from the query or click `Run` to execute the query.
 
 :::tip
-Query results can be transformed using transformations. Read our transformations documentation to see how: [link](/docs/tutorial/transformations)
+Query results can be transformed using Transformation. For more information on transformations, please refer to our documentation at **[link](/docs/tutorial/transformations)**.
 :::
+
+<div style={{textAlign: 'center'}}>
+
+<img className="screenshot-full" src="/img/datasource-reference/rest-api/preview.png" alt="ToolJet - Data source - REST API" />
+
+</div>
+
+## Response types
+
```

Comment:
Should mention this:
if response content-type is image, the response will be base64 string